iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ReferencePriorities.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2023, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.extensions;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.iidm.network.*;
+
+import java.util.*;
+
+/**
+ * ReferencePriorities iIDM extension allows specifying priorities among network equipments terminals for the selection
+ * of a reference bus, for example the angle reference in the context of a load flow.<br/>
+ * The extension is attached to a {@link Connectable}, i.e. any equipment with Terminals.<br/>
+ * A priority level can be defined, as an integer, for each Terminal of the connectable.<br/>
+ * A priority 0 means should not be used.
+ * 1 is highest priority for selection.
+ * 2 is second highest priority, etc ...
+ * @author Damien Jeandemange {@literal <damien.jeandemange at artelys.com>}
+ */
+public interface ReferencePriorities<C extends Connectable<C>> extends Extension<C> {
+
+    String NAME = "referencePriorities";
+    ImmutableMap<IdentifiableType, Integer> DEFAULT_CONNECTABLE_TYPE_PRIORITIES = Maps.immutableEnumMap(
+            ImmutableMap.<IdentifiableType, Integer>builder()
+                    .put(IdentifiableType.GENERATOR, 1)
+                    .put(IdentifiableType.LOAD, 2)
+                    .put(IdentifiableType.BUSBAR_SECTION, 3)
```

Comment:
This is unexpected: I think that bus bar section must have a default priority higher than load, no? In OLF, we have a slack bus selector that just looks at the more netted bus.

Reply:
I think I will just remove these ReferencePriorities.get(network) methods. E.g. for OLF, best to manage this within OLF working on an LfNetwork. Here it works not looking at components, so it is mostly useless / unuseful code.